### DISCUSSION TITLE
OTA updater not updating

### DISCUSSION THREAD

**Discussion by DaveGalbraith** [technyon/nuki_hub#534] [Q&A] [answered]
I'm trying to update to the beta version (from v9.0) and I'm having trouble in that it never seems to work (I have previously tried to update to the latest release and to various other versions trying to debug an issue I had). I've also tried the manual update options, 

Whichever version I start from and aim for I always end up back where I started (I get messages about rebooting etc so something is happening I'm just not sure what). 

Should this process be as straight forward as it appears (hit the big green or yellow button and wait) or are there some special techniques required.

**Comment by iranl** [comment, marked as answer]
Yes this process should be as straight forward as it appears.

It is a pretty recent feature though that has been developed a lot between 9.0 and current master and I am not certain that updating from 9.00 to latest master will work in all cases.

You can flash latest beta using a build artifact https://github.com/technyon/nuki_hub/actions/runs/11979793322 for your device.
Or wait for final release of 9.02 and update using webflash

**Reply by DaveGalbraith** [reply]
That's great, thanks for that. The key for me was to use the updater when in v9.01. 

However, now that I have 9.02-beta9 installed I am still having trouble with controlling the keypad using json, The issue I have now is that commandResultJson returns noValidCodeSet, and I not sure which code is being referred to, so it's a little tricky to debug (perhaps requires a new discussion topic)

**Reply by DaveGalbraith** [reply]
Ignore this latter point I figured out what the code (it's the keypad code), so I've sorted that out.
